django/contrib/postgres/fields/array.py

```diff
@@ -84,7 +84,14 @@ def db_type(self, connection):
 
     def get_db_prep_value(self, value, connection, prepared=False):
         if isinstance(value, list) or isinstance(value, tuple):
-            return [self.base_field.get_db_prep_value(i, connection, prepared=False) for i in value]
+            prepared_values = []
+            for inner_value in value:
+                if hasattr(inner_value, 'resolve_expression'):
```

Comment:
I agree, this is a left-over part of that work. Having the code here doesn't change any error reporting, and it's relatively trivial, so I've removed it on the basis that stale code is bad.